bloom-housing/bloom
# feat: lottery status (released to partners / retracted)

This PR addresses [#(4057)](https://app.zenhub.com/workspaces/bloom-5dc32d7144bd400001315dac/issues/gh/bloom-housing/bloom/4057)


## Description

Create lottery status enums and status change endpoint. Setup business logic for releasing to partners and retracting actions.

To understand how the status flow works, check out this [Miro board](https://miro.com/app/board/uXjVK4sw5X4=/)

## How Can This Be Tested/Reviewed?

Run migration and run backend in the terminal. Use the swagger open api for the following flows.
Flow 1: Login as an admin, attempt to change the status of a active listing to releasedToPartners, you should receive an error since listing is not closed.
Flow 2: Login as an admin, attempt to change the status of a closed listing to releasedToPartners, you should receive an error since it must be in approved.
Flow 3: Change the status manually in the db to approved, login as an admin, attempt to change the status of a closed listing to releasedToPartners, confirm change was successful, attempt to change status to `ran` (this is retracting the lottery), confirm success.
Flow 4: Attempt flow 3 as a partner or juris admin, confirm you receive permission errors.

## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
